### ISSUE TITLE
Crash on sendTagsToServer since 2.4.1

### ISSUE THREAD

**Issue by rivera-ernesto** [OneSignal/OneSignal-iOS-SDK#211]
![2017-03-20 15 55 56](https://cloud.githubusercontent.com/assets/1261982/24119727/9994eb4e-0d88-11e7-99c0-9a5fb826fd1d.png)

**Comment by allaire** [comment]
You should igve a bi tmore info, but personally I'm using sendTags and it works well on latest version:

```swift
        OneSignal.sendTags([
            "userId": session.user.id,
            "firstName": session.user.profile.firstName,
            "lastName": session.user.profile.lastName,
            "deviceLanguage": preferredLocale(),
            "highestRole": session.user.highestRole 
        ])
```

**Comment by jkasten2** [comment]
@rivera-ernesto @chrisfullscreen We haven't seen a crash with `sendTags(s)` however we just fixed an issue where setting them before the the device registered was omitting them in 2.4.1. I just released a 2.4.3 SDK update to fix this case. Can you try seeing if you can reproduce the issue in the latest SDK?

If you still see an issue can you also include all code using `OneSignal` and where it is being called so we can reproduce the issue?

**Comment by rwinzhang** [comment]
I am also experiencing crash, even on 2.5.2 version, when calling OneSignal.deleteTags (it also call OneSignal.sendTags inside). Therefore I downgrade to version 2.4.0 and the crash never happen again.

Here is the logs:
```
[_SwiftTypePreservingNSNumber isEqualToString:]: unrecognized selector sent to instance 0x17023c920
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[_SwiftTypePreservingNSNumber isEqualToString:]: unrecognized selector sent to instance 0x17023c920'
*** First throw call stack:
(0x18e40afd8 0x18ce6c538 0x18e411ef4 0x18e40ef4c 0x18e30ad2c 0x100865224 0x100864db4 0x100418a90 0x1000f2b20 0x100864a08 0x1008794d4 0x10087908c 0x18eaf3630 0x18eed9804 0x18ee1e760 0x18ee0eb18 0x18eedbba0 0x103ef9a10 0x103f072e8 0x103efd634 0x103f09630 0x103f0939c 0x18d4cb1d0 0x18d4cad7c)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by top2tech** [comment]
Same problem here with 2.5.0,  it's intermittent. I may have to stop using OneSignal if it's not fixed before deadline.

Thread 1Queue : com.apple.main-thread (serial)
#0	0x000000018cc9e6d0 in CFRelease ()
#1	0x000000018d4a8190 in +[NSURLConnection sendAsynchronousRequest:queue:completionHandler:] ()
#2	0x000000010032aba0 in +[OneSignalHelper enqueueRequest:onSuccess:onFailure:isSynchronous:] at /Users/Kasten/Documents/OneSignal/OneSignal-iOS-SDK6/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:956
#3	0x000000010032a76c in +[OneSignalHelper enqueueRequest:onSuccess:onFailure:] at /Users/Kasten/Documents/OneSignal/OneSignal-iOS-SDK6/iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m:938
#4	0x0000000100315ae8 in +[OneSignal sendTagsToServer] at /Users/Kasten/Documents/OneSignal/OneSignal-iOS-SDK6/iOS_SDK/OneSignalSDK/Source/OneSignal.m:647
#5	0x000000018d8c646c in __NSFireDelayedPerform ()
#6	0x000000018cd771d8 in __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ ()
#7	0x000000018cd76eec in __CFRunLoopDoTimer ()
#8	0x000000018cd767a8 in __CFRunLoopDoTimers ()
#9	0x000000018cd743a4 in __CFRunLoopRun ()
#10	0x000000018cca22b8 in CFRunLoopRunSpecific ()
#11	0x000000018e756198 in GSEventRunModal ()
#12	0x0000000192ce97fc in -[UIApplication _run] ()
#13	0x0000000192ce4534 in UIApplicationMain ()

**Comment by jkasten2** [comment]
@rwinzhang We have fixed the `isEqualToString:` error in commit 2d0552c9c09a3cbf5e3e1c768c227501aa90c61b.

@rivera-ernesto @chrisfullscreen @top2tech I have changed the threading on this which I believe should fix the `CFRelease` release crash in commit d4698946681e394daa55fe77ab9d6d05a5c43aac.

I'll be releasing a 2.5.3 later today which will include these commits.
